proto/vtgate.proto

```diff
@@ -282,6 +310,36 @@ message StreamExecuteResponse {
   Session session = 2;
 }
 
+// StreamExecuteMultiRequest is the payload to StreamExecuteMulti.
+message StreamExecuteMultiRequest {
+  // caller_id identifies the caller. This is the effective caller ID,
+  // set by the application to further identify the caller.
+  vtrpc.CallerID caller_id = 1;
+
+  // sql contains the set of queries to execute.
+  string sql = 2;
+
+  // session carries the session state.
+  Session session = 3;
+}
+
+// StreamExecuteMultiResponse is the returned value from StreamExecuteMulti.
+message StreamExecuteMultiResponse {
+  // result contains the result data.
+  // The first value contains only Fields information.
+  // The next values contain the actual rows, a few values per result.
+  query.ResultWithError result = 1;
```

Comment:
We can change this behaviour, we do not need a separate packet for fields. Let's not add that comment.

Reply:
Okay 👍